yjit/yjit.mk

```diff
@@ -40,8 +40,37 @@ $(YJIT_LIBS): $(YJIT_SRC_FILES)
 else
 endif
 
-# Put this here instead of in common.mk to avoid breaking nmake builds
-miniruby$(EXEEXT): $(YJIT_LIBS)
+yjit-libobj: $(YJIT_LIBOBJ)
+
+YJIT_LIB_SYMBOLS = $(YJIT_LIBS:.a=).symbols
+$(YJIT_LIBOBJ): $(YJIT_LIBS)
+	$(ECHO) 'partial linking $(YJIT_LIBS) into $@'
+ifneq ($(findstring linux,$(target_os)),)
+	$(Q) $(LD) -r -o $@ --whole-archive $(YJIT_LIBS)
+	$(Q) $(OBJCOPY) --wildcard --keep-global-symbol='$(SYMBOL_PREFIX)rb_*' $(@)
+else ifneq ($(findstring darwin,$(target_os)),)
+	$(Q) $(CC) -nodefaultlibs -r -o $@ -exported_symbols_list $(YJIT_LIB_SYMBOLS) $(YJIT_LIBS)
+else
+	false
```

Comment:
We seem to support (non-Darwin) BSD in addition to Linux and macOS https://github.com/ruby/ruby/blob/edd00697a88892b477b498b796d9f62f9ec84e28/configure.ac#L3788-L3789. Is it okay that it doesn't seem to be covered here?

Reply:
Ah yes, that's a good point. BSDs use bsdmake so they'll use `yjit/not_gmake.mk`. In any case, they don't work under this new setup. I'll make a note to followup on this.

Reply:
Added [857b3da](https://github.com/ruby/ruby/pull/7115/commits/857b3da4ccfb52b4e00e16042eb3dc719dc2ee0c). I tested on FreeBSD 13 since they ship an AWS image so was easy to spin an instance up and test. @jeremyevans if you have time, could you try building on OpenBSD with that commit works?